sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -386,10 +364,16 @@ ItemDeserializer getItemDeserializer() {
                         boolean isNoChangesResponse = isChangeFeed ?
                             ModelBridgeInternal.getNoCHangesFromFeedResponse(page)
                             : false;
-                        List<Mono<byte[]>> byteArrayMonoList =
-                            page.getResults().stream().map(jsonNode -> decryptResponse((ObjectNode) jsonNode)).collect(Collectors.toList());
+                        List<Mono<JsonNode>> byteArrayMonoList =
+                            page.getResults().stream().map(jsonNode -> decryptResponseNode(jsonNode)).collect(Collectors.toList());
                         return Flux.concat(byteArrayMonoList).map(
-                            item -> getItemDeserializer().parseFrom(classType, item)
+                            item -> {
+                                if (item.isValueNode()) {
+                                    return (T)item;
+                                } else {
+                                    return getItemDeserializer().parseFrom(classType, EncryptionUtils.serializeJsonToByteArray(EncryptionUtils.getSimpleObjectMapper(), item));
```

Comment:
Wouldn't this fail, since we are serializing json to byte array, but the return type is JsonNode?

Reply:
@aayush3011  can we directly convert json to class type ? It may avoid conversion to byte array first

Reply:
It should not, the return type here is the ItemDeserializer, the bytes are parsed correctly in the given classType.